### PR TITLE
Allow to narrow down choices in an Enum field

### PR DESCRIPTION
**Pull request by wojcikstefan**
This PR:
* Allows you to only consider a subset of a specific enum's options valid.
* Installs `enum34` in tests so that we can test the `Enum` field in Python v2.
* Is backward-compatible, unless someone initialized `Enum()` using positional arguments (highly unlikely).

### CODE REVIEW

File: cleancat/base.py [hunk 1]
```diff
@@ -491,12 +492,19 @@ class Enum(Choices):
     """
     Like Choices, but expects a Python 3 Enum.
     """
+    def __init__(self, enum_cls, choices=None, **kwargs):
```
Comment (thomasst) [inline cleancat/base.py]: Should add a docstring since it's a bit confusing if you don't follow the code.

Comment (thomasst) [reply cleancat/base.py]: Another way would be to allow you to do `Enum([MyChoices.A, MyChoices.B])`, and derive `enum_cls` via `choices[0].__class__` if `inspect.isclass(choices)` returned false. That way you wouldn't need two arguments and you wouldn't have to specify the class type explicitly.

File: cleancat/base.py [hunk 2]
```diff
@@ -492,11 +493,20 @@ class Enum(Choices):
...
+        * an enum.Enum class (in which case all of its values will become
+          valid choices),
+        * a list containing a subset of the enum's choices (e.g.
+          `[SomeEnumCls.OptionA, SomeEnumCls.OptionB]`).
+        """
+        is_cls = inspect.isclass(choices)
+        if is_cls:
+            self.enum_cls = choices
+        else:
+            self.enum_cls = choices[0].__class__
```
Comment (thomasst) [inline cleancat/base.py]: What do we do if it's an empty list? If it's not a required field it should probably just work. Might want to add a note to the docstring that all choices have to belong to the same enum.

Comment (wojcikstefan) [reply cleancat/base.py]: Done in ddc3cbe.

File: cleancat/base.py [hunk 3]
```diff
@@ -491,12 +493,31 @@ class Enum(Choices):
...
+    def __init__(self, choices, **kwargs):
+        """Initialize the Enum field.
+
+        The `choices` param can be either:
+        * an enum.Enum class (in which case all of its values will become
+          valid choices),
+        * a list containing a subset of the enum's choices (e.g.
+          `[SomeEnumCls.OptionA, SomeEnumCls.OptionB]`). You should provide
+          more than one choice in this list and *all* of the choices should
+          belong to the same enum class.
```
Comment (thomasst) [inline cleancat/base.py]: I like to use "must" or "need to" since it's stronger (you used "should" twice here).

File: setup.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 test_requirements = install_requirements + [
     'pytest',
     'coverage',
+    'enum34',
```
Comment (thomasst) [inline setup.py]: Does this do the right thing in Python 3 (i.e., we keep using the native enum?)

**Pull request merged**